src/cloudai/systems/slurm/slurm_command_gen_strategy.py

```
@@ -105,6 +106,149 @@ def container_mounts(self) -> list[str]:
+    def _gen_hf_model_staging_command(
+        self,
+        models: Iterable[HFModel],
+        local_hf_home_path: Path,
+        node_spec: tuple[int, list[str]] | None = None,
+    ) -> str:
+        """Generate a host-side Slurm step that stages HF models on every compute node."""
+        models_to_stage = sorted(set(models), key=lambda model: model.model_name)
+        shared_hf_home = self.system.hf_home_path.absolute()
+        local_hf_home = local_hf_home_path.absolute()
+        if shared_hf_home == local_hf_home:
+            raise ValueError("hf_local_home_path must be different from hf_home_path")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)